### PR TITLE
refactor: move to /authorization/sources/built_in_database/rules

### PR DESCRIPTION
**Pull request by sstrigler**
Fixes [EMQX-8372](https://emqx.atlassian.net/browse/EMQX-8372)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [x] For internal contributor: there is a jira ticket to track this change
- [x] Coordinate with Dashboard-Team

### CODE REVIEW

**Comment by sstrigler** [comment]
@lafirest could you help with translations please?

File: apps/emqx_authz/i18n/emqx_authz_api_mnesia_i18n.conf [hunk 1]
```diff
@@ -1,100 +1,100 @@
...
     }
   }
 
-  purge_all_delete {
+  rules_delete {
     desc {
-      en: """Purge all records for username/clientid/all"""
-      zh: """清除所有内置数据库中的规则, 用户名/客户端标识符/所有"""
+      en: """Delete all rules for all 'users', 'clients' and 'all'"""
+      zh: """[FIXME]清除所有内置数据库中的规则, 用户名/客户端标识符/所有"""
```
Comment (lafirest) [inline apps/emqx_authz/i18n/emqx_authz_api_mnesia_i18n.conf:97]: 清除内置数据库中的所有类型（'users' 、'clients' 、'all'）的所有规则

File: changes/v5.0.13-zh.md [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 - 重构:  `/users/{name}/change_pwd` 的请求方式从 `PUT` 改为了 `POST` [#9533](https://github.com/emqx/emqx/pull/9533)。
 
+- [FIXME] - Refactor `/authorization/sources/built_in_database/` by adding `rules/` to the path [#9569](https://github.com/emqx/emqx/pull/9569)。
```
Comment (lafirest) [inline changes/v5.0.13-zh.md:13]: 重构 `/authorization/sources/built_in_database/`  接口，将 `rules/` 添加到了其路径中 [#9569](https://github.com/emqx/emqx/pull/9569)。

File: apps/emqx_authz/src/emqx_authz_api_mnesia.erl [hunk 3]
```diff
@@ -70,19 +70,19 @@ api_spec() ->
 
 paths() ->
     [
-        "/authorization/sources/built_in_database/username",
-        "/authorization/sources/built_in_database/clientid",
-        "/authorization/sources/built_in_database/username/:username",
-        "/authorization/sources/built_in_database/clientid/:clientid",
-        "/authorization/sources/built_in_database/all",
-        "/authorization/sources/built_in_database/purge-all"
+        "/authorization/sources/built_in_database/rules/users",
```
Comment (HJianBo) [inline apps/emqx_authz/src/emqx_authz_api_mnesia.erl:73]: 🤔  We should not delete the existing interface directly, which will bring compatibility problems.
IMO, we should mark the existing interfaces as deprecated but still accessible. and add new interfaces under the `rules` path?

Comment (id) [reply apps/emqx_authz/src/emqx_authz_api_mnesia.erl:73]: @HJianBo the background of this task is to make as many breaking changed in our API as possible before the e5.0.0 release.

**Comment by id** [comment]
@sstrigler please rebase on latest master to unblock windows and macos builds

File: apps/emqx_authz/src/emqx_authz_api_mnesia.erl (hunk 3 above)
Comment (HJianBo) [reply apps/emqx_authz/src/emqx_authz_api_mnesia.erl:73]: Got it. thanks for your explanation

**Comment by thalesmg** [comment]
I think the base branch needs to merge or rebase with master, because the way changelogs are constructed has changed: https://github.com/emqx/emqx/pull/9682

**Comment by sstrigler** [comment]
> I think the base branch needs to merge or rebase with master, because the way changelogs are constructed has changed: #9682

My idea is to merge all things into `dev/api-refactor` first and then fix it there before we merge the integration branch.

File: apps/emqx_authz/i18n/emqx_authz_api_mnesia_i18n.conf [hunk 4]
```diff
@@ -1,100 +1,100 @@
...
-      en: """
-Create/Update the list of rules for all.
-Set a empty list to clean up rules
-"""
-      zh: """
-创建/更新 为所有客户端启用的规则列表。
-设为空列表以清楚所有规则
-"""
+      en: """Delete rules for 'all'"""
+      zh: """[FIXME]"""
```
Comment (thalesmg) [inline apps/emqx_authz/i18n/emqx_authz_api_mnesia_i18n.conf:90]: Reminder: needs translation.

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_authz/i18n/emqx_authz_api_mnesia_i18n.conf (hunk 4 above)
Comment (sstrigler) [reply apps/emqx_authz/i18n/emqx_authz_api_mnesia_i18n.conf:90]: @lafirest could you help out here please?

Comment (lafirest) [reply apps/emqx_authz/i18n/emqx_authz_api_mnesia_i18n.conf:90]: 删除 `all` 规则

File: changes/v5.0.13-zh.md [hunk 5]
```diff
@@ -10,11 +10,15 @@
...
 - 重构:  `/users/{name}/change_pwd` 的请求方式从 `PUT` 改为了 `POST` [#9533](https://github.com/emqx/emqx/pull/9533)。
 
+
 - 规则引擎支持压缩函数 `zip`, `gzip`, `zip_compress` 及对应的解压缩函数 [#9573](https://github.com/emqx/emqx/pull/9573)。
 
 - 现在，`PUT /authenticator/:id` 将会返回 204 而不再是 200 [#9434](https://github.com/emqx/emqx/pull/9434/)。
 
-- [FIXME] `/bridges_probe` API 端点用于测试创建新数据桥的参数 [#9585](https://github.com/emqx/emqx/pull/9585)。
+
+- `/bridges_probe` API 端点用于测试创建新数据桥的参数 [#9585](https://github.com/emqx/emqx/pull/9585)。
```
[the remark this answers is not in the document]
Comment (lafirest) [reply changes/v5.0.13-zh.md:21]: 添加新 API 接口 `/bridges_probe` 用于测试创建桥接的参数是否可用。

**Pull request merged**